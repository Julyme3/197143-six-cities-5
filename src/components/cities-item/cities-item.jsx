import React, {memo} from "react";
import PropTypes from "prop-types";

const CitiesItem = ({city, activeCity, onClickHandler}) => {
  return (
    <li
      onClick={() => {
        onClickHandler(city);
      }}
      className="locations__item"
    >
      <a className={`locations__item-link tabs__item ${city === activeCity ? `tabs__item--active` : ``}`} href="#">
        <span>{city}</span>
      </a>
    </li>
  );
};

CitiesItem.propTypes = {
  city: PropTypes.string.isRequired,
  onClickHandler: PropTypes.func.isRequired,
  activeCity: PropTypes.string.isRequired,
};

export {CitiesItem};
export default memo(CitiesItem);
