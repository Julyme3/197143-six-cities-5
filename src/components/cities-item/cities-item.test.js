import React from "react";
import renderer from "react-test-renderer";
import {CitiesItem} from "./cities-item";

const noop = () => {};

describe(`CitiesItem render`, () => {
  it(`Current city is active`, () => {
    const tree = renderer
      .create(
          <CitiesItem
            onClickHandler={noop}
            city="Amsterdam"
            activeCity="Amsterdam"
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Current city is not active`, () => {
    const tree = renderer
      .create(
          <CitiesItem
            onClickHandler={noop}
            city="Amsterdam"
            activeCity="Paris"
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
