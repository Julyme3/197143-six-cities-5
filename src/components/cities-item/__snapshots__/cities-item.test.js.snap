// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesItem render Current city is active 1`] = `
<li
  className="locations__item"
  data-name="Amsterdam"
  onClick={[Function]}
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;

exports[`CitiesItem render Current city is not active 1`] = `
<li
  className="locations__item"
  data-name="Amsterdam"
  onClick={[Function]}
>
  <a
    className="locations__item-link tabs__item "
    href="#"
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;
